src/index.js

```diff
@@ -42,6 +42,38 @@ export function getSimulator(
   };
 }
 
+export function getService<TDeps, TService>(
+  app: FusionApp,
+  plugin: FusionPlugin<TDeps, TService>
+): TService {
+  const token: Token<TService> = createToken('service-helper');
+
+  let extractedService = null;
+  app.register(token, plugin);
+  app.register(
+    createPlugin({
+      deps: {service: token},
+      provides: ({service}) => {
+        extractedService = service;
+      },
+    })
+  );
+
+  try {
+    app.resolve();
```

Comment:
Ugh, this is pretty frustrating.  We can't resolve an `app` twice, which means if we use only the public API for `FusionPlugin` we are destructively modifying `app` and are limited to a single `getService` call.

Thoughts?

Reply:
Limiting tests to getting the service just a single time seems alright to me?

Reply:
I feel like getService should instead be a fixture factory, e.g.

    const service = getService(() => {
      const app = new App()
      app.register(plugin)
      return app
    })

Or possibly

    const service = getService(() => new App(), plugin)